### ISSUE TITLE
Incompatible with Drupal 8.4

### ISSUE THREAD

**Issue by mxhCodes** [valiton/harbourmaster-sso-drupal8-plugin#2]
It seems that the upgrade of Drupal's libraries break the functionality of the current Harbourmaster integration. For example, Drupal 8.4 uses a different major version of jQuery than Drupal 8.3 does.

We receive the following errors on the login page since the upgrade to Drupal 8.4:

```
This page was loaded non-securely in an incognito mode browser. A warning has been added to the URL bar. For more information, see https://goo.gl/y8SRRv.
libs.js:2611 Uncaught TypeError: Cannot set property 'dragstart' of undefined
    at libs.js:2611
    at Array.forEach (<anonymous>)
    at n (libs.js:2605)
    at libs.js:2611
    at t (libs.js:2603)
    at t (libs.js:2603)
    at t (libs.js:2603)
    at t (libs.js:2603)
    at libs.js:2613
    at libs.js:2613
app.js:22 Uncaught TypeError: Cannot set property 'confirmation/confirmation' of undefined
    at app.js:22
jquery.min.js?v=3.2.1:2 jQuery.Deferred exception: Cannot read property 'translations' of undefined TypeError: Cannot read property 'translations' of undefined
    at HTMLDocument.<anonymous> (http://msg.dev/modules/contrib/harbourmaster/js/loginLauncher.js?oyuiz2:318:90)
    at j (http://msg.dev/core/assets/vendor/jquery/jquery.min.js?v=3.2.1:2:29999)
    at k (http://msg.dev/core/assets/vendor/jquery/jquery.min.js?v=3.2.1:2:30313) undefined
r.Deferred.exceptionHook @ jquery.min.js?v=3.2.1:2
jquery.min.js?v=3.2.1:2 Uncaught TypeError: Cannot read property 'translations' of undefined
    at HTMLDocument.<anonymous> (loginLauncher.js?oyuiz2:318)
    at j (jquery.min.js?v=3.2.1:2)
    at k (jquery.min.js?v=3.2.1:2)
```

**Comment by mxhCodes** [comment]
Example page: https://www.mein-schoener-garten.de/harbourmaster/login

**Comment by NoDiskInDriveA** [comment]
This seems to be an incompatibility between jQuery3 and the underlying EmberJS 1.7.1.

Since the example page currently redirects, I had to debug on the working version of the login page, but I found out the following:
When interacting with jQuery, EmberJS tries to set properties on jQuery.event.fixHooks, which has been removed (https://jquery.com/upgrade-guide/3.0/#breaking-change-jquery-event-props-and-jquery-event-fixhooks-removed). This unchecked access raises the error which leaves EmberJS incomplete, and triggers the other errors down the line.
Additionally, there's a minor bug when checking the EmberJS translations property.

As this is a problem with the EmberJS application and not with the Drupal plugin itself, I will have to delegate this issue to the widget developers.
